src/engine/RenderSystem.cpp

```diff
@@ -0,0 +1,145 @@
+#include "engine/RenderSystem.hpp"
+
+#include "util/GLDebug.hpp"
+#include <array>
+
+namespace lei3d {
+
+void RenderSystem::initialize(int width, int height) {
+    scwidth = width;
+    scheight = height;
+
+    forwardShader = Shader("./data/shaders/forward.vert", "./data/shaders/forward.frag");
+    postprocessShader = Shader("./data/shaders/screenspace_quad.vert", "./data/shaders/postprocess.frag");
+
+    glGenVertexArrays(1, &dummyVAO);
+
+    glGenFramebuffers(1, &FBO);
+    glBindFramebuffer(GL_DRAW_FRAMEBUFFER, FBO);
+
+    glGenTextures(1, &rawTexture);
+    glGenTextures(1, &saturationMask);
+    glGenTextures(1, &depthStencilTexture);
+    glGenTextures(1, &finalTexture);
+
+    // lighting pass
+    glBindTexture(GL_TEXTURE_2D, rawTexture);
+    glTexImage2D(GL_TEXTURE_2D, 0, GL_RGB32F, width, height, 0, GL_RGB, GL_FLOAT, nullptr);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+    glFramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, GL_TEXTURE_2D, rawTexture, 0);
```

Comment:
Do these load the output of the color attachments to the texture that you minded in glBindTexture?

Reply:
Or does it also attach it the texture to the Frame buffer object?